akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala

```diff
@@ -53,13 +54,17 @@ object ClusterEvent {
 
   /**
    * Current snapshot state of the cluster. Sent to new subscriber.
+   *
+   * @param leader this field should not be used if cluster teams are used as in that case there is no single leader
+   *               for the entire cluster. In that case you should instead use [[CurrentClusterState#leader(String)]]
```

Comment:
Perhaps this can be simplified so that `leader[Option]` means the leader in the own team.
`CurrentClusterState` is accessed via local `Cluster(system).state` or via the cluster event subscriptions, which are also intended to be local.